dashboard/src/components/Catalog/CatalogItem.tsx

```diff
@@ -37,18 +38,19 @@ function trimDescription(desc: string): string {
 
 const CatalogItem: React.SFC<ICatalogItemProps> = props => {
   const { item } = props;
-  const { icon, name, repoName, version, description, type, namespace, id, csv } = item;
+  const { icon, name, repo, version, description, type, namespace, id, csv } = item;
   const iconSrc = icon || placeholder;
   let link;
   let tag1;
   let subIcon;
   if (type === "chart") {
     tag1 = (
-      <Link className="ListItem__content__info_tag_link" to={`/catalog/${repoName}`}>
-        {repoName}
+      <Link className="ListItem__content__info_tag_link" to={`/catalog/${repo?.name}`}>
+        {repo?.name}
       </Link>
     );
-    link = `/charts/${repoName}/${name}`;
+    const chartsSegment = namespace !== repo?.namespace ? "global-charts" : "charts";
+    link = `/ns/${namespace}/${chartsSegment}/${repo?.name}/${name}`;
```

Comment:
feel free to ignore this suggestion but as a matter of style I would prefer the global route to be something like `ns/foo/charts/bitnami/wordpress?global=true`. Also, with that you don't need to define two different routes for the same container.

Reply:
That would be using the query params for hierarchical data, rather than for non-hierarchical data (sorting / filtering), which isn't great, imo. That is, the Uniform Resource Identifier should identify the hierarchical resource using the path only (see https://tools.ietf.org/html/rfc3986#section-3.4, or better explained specifically for restful apis at https://stackoverflow.com/questions/30967822/when-do-i-use-path-params-vs-query-params-in-a-restful-api ).

The easy test here is, does the path without a query params identify a resource? This is why either `global` or the explicit chart namespace is required in the URI IMO, to identify the resource unambiguously.